### ISSUE TITLE
Advanced conditional coverage

### ISSUE THREAD

**Issue by gyandeeps** [gotwarlost/istanbul#311]
Just curious, is there a way to know whether all the conditions got evaluated both ways as `true` and `false` inside conditional logics.

Let me explain this with an example:

```.js
//Main file
function awesome(a, b){
    if(a && b){
        return true;
    }
    return false;
}
```

```.js
//Test file
descrive("test", function(){
    it("some", function(){
        expect(awesome(true, true)).toBe(true);
        expect(awesome(false, true)).toBe(false);
        expect(awesome(true, true)).toBe(true);
        expect(awesome(false, true)).toBe(false);
    });
});
```

Is there a way that code coverage can say that `b` condition was always tested for `true` and was never tested for a `false` evaluation?

**Comment by gotwarlost** [comment]
No, there is no way to do this. I mean, it can be done, but _probably_ not as a small delta over the current istanbul design.

**Comment by gyandeeps** [comment]
@gotwarlost Can you please give me the area where you think this changes needs to be made?
Just as a starting point as I am curious and I want to play around with this library a little bit.

CC: @davglass
